src/crush/CrushWrapper.cc

```diff
@@ -1540,6 +1572,38 @@ void CrushWrapper::decode(bufferlist::iterator& blp)
 	class_rname[c.second] = c.first;
       ::decode(class_bucket, blp);
       cleanup_classes();
+
```

Comment:
can you add a } if (!blp.end()) { here?  i have the class code deployed on a test cluster.